bindings/go/repositories/componentrepository/fallback/repository.go

```diff
@@ -0,0 +1,344 @@
+func (f *FallbackRepository) AddComponentVersion(ctx context.Context, descriptor *descriptor.Descriptor) error {
+	repos := f.RepositoriesForComponentIterator(ctx, descriptor.Component.Name)
+	for repo, err := range repos {
+		if err != nil {
+			return fmt.Errorf("getting repository for component %s failed: %w", descriptor.Component.Name, err)
+		}
+		return repo.AddComponentVersion(ctx, descriptor)
+	}
+	return fmt.Errorf("no repository found for component %s to add version", descriptor.Component.Name)
+}
+
+// Deprecated: FallbackRepository is an implementation for the deprecated config
+// type "ocm.config.ocm.software/v1". This concept of fallback resolvers is deprecated
+// and only added for backwards compatibility.
+// New concepts will likely be introduced in the future (contributions welcome!).
+//
+// GetComponentVersion iterates through all resolvers with matching component prefix in
+// the order of their priority (higher priorities first) and attempts to
+// retrieve the component version from each repository.
+func (f *FallbackRepository) GetComponentVersion(ctx context.Context, component, version string) (*descriptor.Descriptor, error) {
+	repos := f.RepositoriesForComponentIterator(ctx, component)
+	for repo, err := range repos {
+		if err != nil {
+			return nil, fmt.Errorf("getting repository for component %s failed: %w", component, err)
+		}
+		desc, err := repo.GetComponentVersion(ctx, component, version)
+		if err != nil {
+			if errors.As(err, new(*componentrepository.NotFoundError)) {
+				slog.DebugContext(ctx, "component version not found in repository", "realm", componentrepository.Realm, "repository", repo, "component", component, "version", version)
+				continue // try the next repository
+			}
+			return nil, fmt.Errorf("getting component version %s/%s from repository %v failed: %w", component, version, repo, err)
+		}
+		return desc, nil
+	}
+	return nil, fmt.Errorf("component version %s/%s not found in any repository", component, version)
+}
+
+// Deprecated: FallbackRepository is an implementation for the deprecated config
+// type "ocm.config.ocm.software/v1". This concept of fallback resolvers is deprecated
+// and only added for backwards compatibility.
+// New concepts will likely be introduced in the future (contributions welcome!).
+//
+// ListComponentVersions accumulates a deduplicated list of the versions of the
+// given component from all repositories specified by resolvers with a matching
+// component prefix in the order of their priority (higher priorities first).
+func (f *FallbackRepository) ListComponentVersions(ctx context.Context, component string) ([]string, error) {
+	repos := f.RepositoriesForComponentIterator(ctx, component)
+
+	var errGroup errgroup.Group
+	var versionsMu sync.Mutex
+	accumulatedVersions := make(map[string]struct{})
+
+	errGroup.SetLimit(goruntime.NumCPU())
+
+	for repo, err := range repos {
+		errGroup.Go(func() error {
```

Comment:
initialize the errgroup with errgroup.WithContext then you dont need to manually check for context cancellation